mdata/notifierKafka/notifierKafka.go

```diff
@@ -93,11 +96,33 @@ func (c *NotifierKafka) start() {
 		if err != nil {
 			log.Fatal(4, "kafka-cluster: Failed to get %q duration offset for %s:%d. %q", offsetStr, topic, partition, err)
 		}
-		go c.consumePartition(topic, partition, offset)
+		processBacklog.Add(1)
+		go c.consumePartition(topic, partition, offset, processBacklog)
+	}
+	// wait for our backlog to be processed before returning.  This will block metrictank from consuming metrics until
+	// we have processed old metricPersist messages. The end result is that we wont overwrite chunks in cassandra that
+	// have already been previously written.
+	// We dont wait more then 60seconds for the backlog to be processed.
+	log.Info("kafka-cluster: waiting for metricPersist backlog to be processed.")
+	backlogProcessed := make(chan struct{}, 1)
+	go func() {
+		processBacklog.Wait()
+		backlogProcessed <- struct{}{}
+	}()
+	ticker := time.NewTicker(time.Minute)
+	for {
+		select {
+		case <-ticker.C:
+			log.Warn("kafka-cluster: Processing metricPersist backlog has taken too long, giving up lock.")
```

Comment:
So if this reaches the timeout we only put a warning message in the log and then continue as usual? That means if the timeout gets reached and we're not done processing the backlog yet, it's fine to still just continue starting up as usual. That basically makes the whole timeout only informational for people who're looking at the logs, but otherwise the waiting for a timeout has no function. 
Wouldn't it be more consequent to print a warning like that each minute and then continue waiting for the backlog until it's really processed? Or otherwise, if the waiting isn't really required, why do we wait at all?